pinot-common/src/main/java/com/linkedin/pinot/common/utils/DataTableBuilder.java

```diff
@@ -323,11 +328,15 @@ public void setColumn(int columnIndex, String value) throws Exception {
    * @throws Exception
    */
   public void setColumn(int columnIndex, Object value) throws Exception {
-
-    byte[] bytes = new byte[0];
-    bytes = serializeObject(value);
+    byte[] bytes = dataTableSerDe.serialize(value);
     currentRowData.position(columnOffsets[columnIndex]);
     currentRowData.putInt(variableSizeDataHolder.position());
+
+    // For custom serialization, we need to write the object type as well.
+    if (version == DataTable.Version.V2) {
+      variableSizeDataHolder.add(dataTableSerDe.getObjectType(value).getValue());
```

Comment:
Yes, and the reason being different rows of the same column can potentially have different types.  This will be addressed in the future re-designs/clean-ups of data table.